torch/autograd/grad_mode.py

```diff
@@ -29,13 +38,23 @@ def __exit__(self, *args):
         torch.set_grad_enabled(self.prev)
         return False
 
+    def __call__(self, func):
+        def decorate_no_grad(*args, **kwargs):
```

Comment:
nit: use `functools.wraps` on this please. Also, can we use the context manager style to avoid exceptions in `func`?
```
with self:
  return func(*args, **kwargs)
```

Reply:
wrote it this way to avoid the extra function call, but that makes sense. fixed